Index: src/dhcpc/dhcpc.c
===================================================================
--- src/dhcpc/dhcpc.c   (revision 615)
+++ src/dhcpc/dhcpc.c   (working copy)
@@ -38,6 +38,7 @@
 #include "dhcpc.h"
 #include "../uip/timer.h"
 #include "../uip/pt.h"
+#include "../logger.h"
 
 #define STATE_INITIAL         0
 #define STATE_SENDING         1
@@ -177,6 +178,7 @@
   end = add_end(end);
 
   uip_send(uip_appdata, end - (u8_t *)uip_appdata);
+  uip_log(" dhcp sent discover\n");
 }
 /*---------------------------------------------------------------------------*/
 static void
@@ -193,6 +195,7 @@
   end = add_end(end);
   
   uip_send(uip_appdata, end - (u8_t *)uip_appdata);
+  uip_log(" dhcp sent request\n");
 }
 /*---------------------------------------------------------------------------*/
 static u8_t
@@ -235,6 +238,7 @@
 {
   struct dhcp_msg *m = (struct dhcp_msg *)uip_appdata;
   
+  log_message(" dhcp got packet type %d\n", m->op);
   if(m->op == DHCP_REPLY &&
      memcmp(m->xid, xid, sizeof(xid)) == 0 &&
      memcmp(m->chaddr, s.mac_addr, s.mac_len) == 0) {
Index: src/uip/uip_task.c
===================================================================
--- src/uip/uip_task.c   (revision 615)
+++ src/uip/uip_task.c   (working copy)
@@ -65,11 +65,11 @@
 
     // init uIP
     uip_init();
-    uip_log("uip init");
+    uip_log("uip init\n");
 
     // init ARP cache
       uip_arp_init();
-      uip_log("arp init");
+      uip_log("arp init\n");
 
    dhcpc_init(&mac, 6);
    //telnetd_init();
Index: src/uip/uip-conf.h
===================================================================
--- src/uip/uip-conf.h   (revision 615)
+++ src/uip/uip-conf.h   (working copy)
@@ -107,7 +107,7 @@
  *
  * \hideinitializer
  */
-#define UIP_CONF_BUFFER_SIZE     512
+#define UIP_CONF_BUFFER_SIZE     1500
 
 #define UIP_CONF_EXTERNAL_BUFFER
 
@@ -132,7 +132,7 @@
  * \hideinitializer
  */
 #define UIP_CONF_UDP             1
-
+#define UIP_CONF_BROADCAST            1
 /**
  * UDP checksums on or off
  *
Index: src/nic/enc28j60.h
===================================================================
--- src/nic/enc28j60.h   (revision 615)
+++ src/nic/enc28j60.h   (working copy)
@@ -242,7 +242,7 @@
 //#define RXSTART_INIT      0x46
 //#define RXSTOP_INIT       0x60
 
-#define   MAX_FRAMELEN   512   // maximum ethernet frame length; should match UIP Buffer size
+#define   MAX_FRAMELEN   1500   // maximum ethernet frame length; should match UIP Buffer size
 
 // Ethernet constants
 #define ETHERNET_MIN_PACKET_LENGTH   0x3C